grails-app/services/au/org/emii/portal/BulkDownloadService.groovy

```diff
@@ -145,4 +145,14 @@ class BulkDownloadService {
         zipStream.write bytes, 0, bytes.length
         zipStream.closeEntry()
     }
+
+    public static void copy(final InputStream input, final OutputStream output, final int bufferSize) throws IOException
```

Comment:
If it's a hassle to pull in the library, don't worry too much about it, although I'm always wary of implementing fairly low-level stuff like this ourselves, as I think it's always a bug breeding ground.  p.s. dare I say it, the code conventions are not met in the method, i.e. opening braces are on newlines, no space after `while`).